packages/assets/lib/returnImage.js

```diff
@@ -2,61 +2,78 @@ const sharp = require('sharp')
 const getWidthHeight = require('./getWidthHeight')
 const fileTypeStream = require('file-type-stream').default
 const { PassThrough } = require('stream')
-const pick = require('lodash/pick')
+
+const pipeHeaders = [
+  'Content-Type',
+  'Last-Modified',
+  'cache-control',
+  'expires',
+  'Access-Control-Allow-Credentials',
+  'Access-Control-Allow-Headers',
+  'Access-Control-Allow-Methods',
+  'Access-Control-Allow-Origin',
+]
 
 module.exports = async ({
   response: res,
   stream,
-  headers = {},
+  headers,
   options = {}
 }) => {
-  const { resize, bw } = options
+  const { resize, bw, webp } = options
   let width, height
-  try {
-    if (resize) {
+  if (resize) {
+    try {
       ({ width, height} = getWidthHeight(resize))
+    } catch (e) {
+      res.status(400).end(e.message)
+    }
+  }
+
+  // forward filtered headers
+  if (headers) {
+    for(let key of pipeHeaders) {
+      const value = headers.get(key)
+      if (value) {
+        res.set(key, value)
+      }
     }
-  } catch (e) {
-    res.status(400).end(e.message)
   }
 
   // detect mime
   const passThrough = new PassThrough()
-  const { mime } = await new Promise(resolve => {
-    stream.pipe(fileTypeStream(resolve)).pipe(passThrough)
-  })
-
-  // set headers
-  res.set(pick({
```

Comment:
And here I'd prefer es6 destructuring as it's more fluent to developers. I had to lookup the signature and meaning of lodash's pick.

The whole pipelining stuff is very cool, dude! 🎉